### ISSUE TITLE
System.ArgumentOutOfRangeException

### ISSUE THREAD

**Issue by robson31** [GrooverFromHolland/SimpleBleExample_by_Devicename#2]
Copy/Paste from https://github.com/nkolban/esp32-snippets/issues/546

Thanks, I appreciate you time to write this comment. I'm still on an early stage of understanding how everything works. I've downloaded the complete solution from your repo and ran the code. It crashes on line 87 inside MainPage.xaml.cs:
System.ArgumentOutOfRangeException: 'Specified argument was out of the range of valid values.'

Thanks again for the sample, it's indeed very well commented and very helpful.

Edit: Here's what's showing for services variable:
 | services | {System.__ComObject} | System.Collections.Generic.IReadOnlyList<Windows.Devices.Bluetooth.GenericAttributeProfile.GattDeviceService> {System.__ComObject}

I've made sure I have entered correct MyService_GUID on line 34. Any help much appreciated.
**************************************************************************************************************
I believe the code that I was using on my HUZZAH32 (esp32 dev board) was somehow incorrect. I've tried my luck with some other examples, such as BLE_server and BLE_notify:
https://github.com/nkolban/ESP32_BLE_Arduino/tree/master/examples

With both examples, I get Exception from HRESULT: 0x80650003. It comes from line 133:
`var descriptorWriteResult = await charac.WriteClientCharacteristicConfigurationDescriptorAsync(descriptorValue);`

Here's the console output:
> Device found in 299 ms
Signalstrengt = -48 DBm
Ble address = 53526636485438
Advertisement type = ConnectableUndirected
Connected in 40 ms
Service found in 3 ms
Characteristic found in 65 ms
This characteristic supports reading .
This characteristic supports writing .
Exception thrown: 'System.Exception' in System.Private.CoreLib.ni.dll
An exception of type 'System.Exception' occurred in System.Private.CoreLib.ni.dll but was not handled in user code
The attribute cannot be written. (Exception from HRESULT: 0x80650003)
The program '[16784] SimpleBleExample_by_Devicename.exe' has exited with code -1 (0xffffffff).

**Comment by GrooverFromHolland** [comment]
@robson31 
The attribute cannot be written. (Exception from HRESULT: 0x80650003) means that your Ble-server is not properly configured for subscribing to notifications or indications.  I suggest to change your issue on https://github.com/nkolban/ESP32_BLE_Arduino/tree/master/examples
or start a new issue. At the moment i don't have a esp32 available to setup the example you're using. Mine are all clients and in use.

**Comment by robson31** [comment]
Thank you for your comment. You are exactly right about that, it doesn't work for some examples. The uart test example however works, since it's configured for subscribing to notifications or indications.
I was able to recreate the same issue on BleLab app and I figured out why your example did not work for me. The key is to pair the device in Windows Bluetooth settings before running the example. After I did that it worked flawlessly.
